crates/rspack_plugin_mf/src/container/embed_federation_runtime_module.rs

```diff
@@ -0,0 +1,111 @@
+//! # EmbedFederationRuntimeModule
+//!
+//! Runtime module that wraps the startup function to ensure federation runtime dependencies
+//! execute before other modules. Generates an "oldStartup wrapper" pattern that intercepts
+//! and modifies the startup execution order.
+
+use rspack_cacheable::cacheable;
+use rspack_collections::Identifier;
+use rspack_core::{
+  impl_runtime_module, module_raw, ChunkUkey, Compilation, DependencyId, RuntimeGlobals,
+  RuntimeModule, RuntimeModuleStage,
+};
+use rspack_error::Result;
+
+#[cacheable]
+#[derive(Debug, Default, Clone, Hash, PartialEq, Eq)]
+pub struct EmbedFederationRuntimeModuleOptions {
+  pub collected_dependency_ids: Vec<DependencyId>,
+}
+
+#[impl_runtime_module]
+#[derive(Debug)]
+pub struct EmbedFederationRuntimeModule {
+  id: Identifier,
+  chunk: Option<ChunkUkey>,
+  options: EmbedFederationRuntimeModuleOptions,
+}
+
+impl EmbedFederationRuntimeModule {
+  pub fn new(options: EmbedFederationRuntimeModuleOptions) -> Self {
+    Self::with_default(
+      Identifier::from("webpack/runtime/embed_federation_runtime"),
+      None,
+      options,
+    )
+  }
+}
+
+#[async_trait::async_trait]
+impl RuntimeModule for EmbedFederationRuntimeModule {
+  fn name(&self) -> Identifier {
+    self.id
+  }
+
+  async fn generate(&self, compilation: &Compilation) -> Result<String> {
+    let chunk_ukey = self
+      .chunk
+      .expect("Chunk should be attached to RuntimeModule");
+
+    let chunk = compilation.chunk_by_ukey.expect_get(&chunk_ukey);
+
+    let collected_deps = &self.options.collected_dependency_ids;
+
+    if collected_deps.is_empty() {
+      return Ok("// No federation runtime dependencies to embed.".to_string());
+    }
+
+    let module_graph = compilation.get_module_graph();
+    let mut federation_runtime_modules = Vec::new();
+
+    // Find federation runtime dependencies in this chunk
+    for dep_id in collected_deps.iter() {
+      if let Some(module_dyn) = module_graph.get_module_by_dependency_id(dep_id) {
+        let is_in_chunk = compilation
+          .chunk_graph
+          .is_module_in_chunk(&module_dyn.identifier(), chunk_ukey);
+        if is_in_chunk {
+          federation_runtime_modules.push(*dep_id);
+        }
+      }
+    }
+
+    if federation_runtime_modules.is_empty() {
+      return Ok("// Federation runtime entry modules not found in this chunk.".to_string());
+    }
+
+    // Generate module execution code for each federation runtime dependency
+    let mut runtime_requirements = RuntimeGlobals::default();
+    let mut module_executions = Vec::new();
+
+    for dep_id in federation_runtime_modules {
+      let module_str = module_raw(compilation, &mut runtime_requirements, &dep_id, "", false);
+      module_executions.push(format!("\t\t{}", module_str));
+    }
+
+    // Generate oldStartup wrapper pattern
+    let result = format!(
```

Comment:
patch __webpack_require__.x which is entry startup function. 

first initialize federation, then call previous startup if exists - if it exists, it will be webpack require of entrypoint module. 

If startup doesnt exist in entrypoint bootstrap, like in single runtime chunk, another plugin will manually add require.x() to the entrypoint chunks at the begenning of startup